stl/inc/bit

```diff
@@ -94,19 +96,124 @@ _NODISCARD constexpr _Ty rotr(const _Ty _Val, const int _Rotation) noexcept {
     }
 }
 
-template <class _Ty, enable_if_t<_Is_standard_unsigned_integer<_Ty>, int> _Enabled>
-_NODISCARD constexpr int countl_zero(const _Ty _Val) noexcept {
+// Implementation of popcount without using specialized CPU instructions.
+// Used at compile time and when said instructions are not supported.
+template <class _Ty>
+_NODISCARD constexpr int _Popcount_fallback(_Ty _Val) noexcept {
+    constexpr int _Digits = numeric_limits<_Ty>::digits;
+    // we static_cast these bit patterns in order to truncate them to the correct size
+    _Val = static_cast<_Ty>(_Val - ((_Val >> 1) & static_cast<_Ty>(0x5555'5555'5555'5555ull)));
+    _Val = static_cast<_Ty>((_Val & static_cast<_Ty>(0x3333'3333'3333'3333ull))
+                            + ((_Val >> 2) & static_cast<_Ty>(0x3333'3333'3333'3333ull)));
+    _Val = static_cast<_Ty>((_Val + (_Val >> 4)) & static_cast<_Ty>(0x0F0F'0F0F'0F0F'0F0Full));
+    for (int _Shift_digits = 8; _Shift_digits < _Digits; _Shift_digits <<= 1) {
+        _Val = static_cast<_Ty>(_Val + static_cast<_Ty>(_Val >> _Shift_digits));
+    }
+    // we want the bottom "slot" that's big enough to store _Digits
+    return static_cast<int>(_Val & static_cast<_Ty>(_Digits + _Digits - 1));
+}
+
+#if defined(_M_IX86) || defined(_M_X64)
+
+// TRANSITION, VS 2019 16.8 Preview 1

```

Comment:
Thanks for the TRANSITION comment. One more suggestion/request - we try to make it reasonably clear what should be changed when a TRANSITION comment is finally resolved. Sometimes, simply commenting the affected line is sufficient; for this reason, we generally avoid having TRANSITION comments on isolated lines (not always though). For more complicated scenarios, we sometimes mention what will need to be changed.

In this case, the comment appears above the `extern "C"` block, but only 6/7 of it is TRANSITION - the `extern int __isa_available;` will need to remain (as we otherwise don't have a declaration of it).

Originally, I imagined each of the `__MACHINEMEOW` declarations being commented, although that is admittedly repetitive. How about this?

```suggestion
// TRANSITION, VS 2019 16.8 Preview 1, intrin0.h will declare __lzcnt* and __popcnt*
```

This isn't *very* important because 16.8p1 will be available soon, and we'll still remember what to do here. It's more important for long-term TRANSITION comments where it might be years before we can resolve them. (Similar long-term planning is why we cited bug databases for bug numbers, and also mention bug titles.)